saleor/tests/e2e/orders/test_apply_better_promotion_to_product.py

```diff
@@ -0,0 +1,192 @@
+    create_product_channel_listing,
+    create_product_type,
+    create_product_variant,
+    create_product_variant_channel_listing,
+    get_product,
+)
+from ..promotions.utils import create_promotion, create_promotion_rule
+from ..utils import assign_permissions
+from .utils import draft_order_create, order_lines_create
+
+
+def prepare_product(
+    e2e_staff_api_client,
+    channel_slug,
+    variant_price,
+):
+    channel_data = create_channel(
+        e2e_staff_api_client,
+        slug=channel_slug,
+    )
+    channel_id = channel_data["id"]
+
+    product_type_data = create_product_type(
+        e2e_staff_api_client,
+    )
+    product_type_id = product_type_data["id"]
+
+    category_data = create_category(
+        e2e_staff_api_client,
+    )
+    category_id = category_data["id"]
+
+    product_data = create_product(
+        e2e_staff_api_client,
+        product_type_id,
+        category_id,
+    )
+    product_id = product_data["id"]
+    create_product_channel_listing(e2e_staff_api_client, product_id, channel_id)
+
+    variant_data = create_product_variant(e2e_staff_api_client, product_id)
+    product_variant_id = variant_data["id"]
+
+    create_product_variant_channel_listing(
+        e2e_staff_api_client,
+        product_variant_id,
+        channel_id,
+        variant_price,
+    )
+
+    return product_id, channel_id, product_variant_id
+
+
+def promotion_with_rules(
+    e2e_staff_api_client,
+    promotion_name,
+    discount_type,
+    discount_value,
+    rule_name,
+    channel_id,
+    product_id,
+):
+    promotion = create_promotion(e2e_staff_api_client, promotion_name)
+    promotion_id = promotion["id"]
+
+    promotion_rule = create_promotion_rule(
+        e2e_staff_api_client,
+        promotion_id,
+        discount_type,
+        discount_value,
+        rule_name,
+        channel_id,
+        product_id,
+    )
+    product_predicate = promotion_rule["cataloguePredicate"]["productPredicate"]["ids"]
+    assert promotion_rule["channels"][0]["id"] == channel_id
+    assert product_predicate[0] == product_id
+
+
+@pytest.mark.e2e
+def test_apply_best_promotion_to_product_core_2105(
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_product_types_and_attributes,
+    permission_manage_discounts,
+    permission_manage_orders,
+):
+    # Before
+    channel_slug = "test-channel"
+    variant_price = "30"
+
+    permissions = [
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_product_types_and_attributes,
+        permission_manage_discounts,
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    product_id, channel_id, product_variant_id = prepare_product(
+        e2e_staff_api_client,
+        channel_slug,
+        variant_price,
+    )
+
+    # Create first promotion
+    first_promotion_name = "Promotion Fixed"
```

Comment:
Shouldn't you have more tests with different combinations of promotions? Currently, you have prom1:fixed, and the second one prom2:percentage. We could also test the case when you have prom1:fixed, prom2:fixed and prom1:percentage, prom2:precentage

Reply:
I added parametrization to this test case to check other cases fixed + percentage, percentage + percentage and fixed + fixed